src/azul/terra.py

```diff
@@ -348,6 +358,93 @@ def _repository_endpoint(self, *path: str) -> str:
         return furl(config.tdr_service_url,
                     path=('api', 'repository', 'v1', *path)).url
 
+    def _check_response(self,
+                        endpoint: str,
+                        headers: JSON = None,
+                        params: JSON = None
+                        ) -> MutableJSON:
+        response = self._oauthed_request('GET', endpoint, headers=headers, fields=params)
+        if response.status == 200:
+            return json.loads(response.data)
+        elif response.status == 401:
+            raise self._insufficient_access(endpoint)
+        else:
+            raise RequirementError('Unexpected response from TDR API', response.status)
```

Comment:
It looks to me like the creation of this method is purely a refactoring, in which case maybe it should go in the previous commit.

Reply:
True, but the refactoring was motivated by the addition of the `TDRClient.list_snapshots` method which would have otherwise had a lot of redundant code with `TDRClient._lookup_source`. The refactoring of the scopes and SAM auth make sense as a standalone code improvement but the motivation for this change wouldn't be clear without the implementation of `list_snapshots` so I think it should stay where it is.

Reply:
If I recall correctly you did end up splitting this out. Even still I just noticed a relevant section of the contributing guide so I thought to share it https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#34bloated-diffs

Reply:
I just noticed a relevant section of the contributing guide so I thought to share it https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#34bloated-diffs.

I leave it to you to decide if it applies here.

If you don't want to split it up the changes, try moving the definition of `_check_response()` directly after where it is used. That may also be sufficient to minimize the diff.